build.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+[[ -f /etc/prfile.d/modules.sh ]] && \
+  source /etc/profile.d/modules.sh
+
+builddir=build
+[ -d $builddir ] && rm -rf $builddir; mkdir $builddir
+
+if type module
+then
+  proj_dir=/qfs/projects/exasgd
+  mod_dir=/share/apps
+  cluster=$(uname -n | cut -d'.' -f1 | sed -E 's/[[:digit:]]//g')
+  module purge
+  if [ "$cluster" == "newell" ]
+  then
+    gcc=7.4.0
+    cmake=3.16.4
+    openmpi=3.1.5
+    cuda=10.2
+    magma=2.5.2_cuda10.2
+    metis=5.1.0
+  else
+    gcc=7.3.0
+    cmake=3.15.3
+    openmpi=3.1.3
+    cuda=10.1.243
+    magma=2.5.2_cuda10.2
+    metis=5.1.0
+  fi
+
+  module load gcc/$gcc
+  module load cmake/$cmake
+  module load openmpi/$openmpi
+  module load cuda/$cuda
+  module load magma/$magma
+  module load metis/$metis
+fi
+
+make_flags="-j 8"
+
+declare -a cmake_args=(
+  "-DHIOP_USE_GPU=ON -DHIOP_MAGMA_DIR=$mod_dir/magma/2.5.2/cuda10.2/ \
+    -DHIOP_USE_MPI=ON -DHIOP_WITH_KRON_REDUCTION=ON \
+    -DHIOP_UMFPACK_DIR=$proj_dir/$cluster/suitesparse \
+    -DHIOP_METIS_DIR=$mod_dir/metis/5.1.0"
+  "-DHIOP_USE_MPI=ON -DHIOP_DEEPCHECKS=OFF -DCMAKE_BUILD_TYPE=RELEASE"
+  "-DHIOP_USE_MPI=OFF -DHIOP_DEEPCHECKS=OFF -DCMAKE_BUILD_TYPE=RELEASE"
+  "-DHIOP_USE_MPI=ON -DHIOP_DEEPCHECKS=ON -DCMAKE_BUILD_TYPE=RELEASE"
+  "-DHIOP_USE_MPI=OFF -DHIOP_DEEPCHECKS=ON -DCMAKE_BUILD_TYPE=RELEASE"
+  "-DHIOP_USE_MPI=ON -DHIOP_DEEPCHECKS=OFF -DCMAKE_BUILD_TYPE=DEBUG"
+  "-DHIOP_USE_MPI=OFF -DHIOP_DEEPCHECKS=OFF -DCMAKE_BUILD_TYPE=DEBUG"
+  "-DHIOP_USE_MPI=ON -DHIOP_DEEPCHECKS=ON -DCMAKE_BUILD_TYPE=DEBUG"
+)
```

Comment:
Flags should probably be specific to the cluster where the test is run.

Reply:
Good point